cmd/lifecycle/creator.go

```diff
@@ -292,19 +285,16 @@ func (c *createCmd) platformAPIVersionGreaterThan06() bool {
 
 func (c *createCmd) ReadableRegistryImages() []string {
 	var readableImages []string
-	if c.platformAPIVersionGreaterThan06() {
-		if !c.useDaemon {
-			readableImages = append(readableImages, c.previousImageRef, c.runImageRef)
-		}
+	if !c.useDaemon {
+		readableImages = appendNotEmpty(readableImages, c.previousImageRef, c.runImageRef)
 	}
 	return readableImages
 }
 func (c *createCmd) WriteableRegistryImages() []string {
 	var writeableImages []string
-	if c.platformAPIVersionGreaterThan06() {
-		if !c.useDaemon {
-			writeableImages = append(writeableImages, append([]string{c.cacheImageRef}, c.additionalTags...)...)
-		}
+	writeableImages = appendNotEmpty(writeableImages, c.cacheImageRef)
+	if !c.useDaemon {
+		writeableImages = append(writeableImages, appendNotEmpty([]string{c.outputImageRef}, c.additionalTags...)...)
```

Comment:
Same comment as https://github.com/buildpacks/lifecycle/pull/646/commits/56d14b18dfe67b24d46995d21395e94847e3d59f#r674434602.
I think that a nested `append` is confusing for the reader.

Reply:
I used two appends in the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/7b1d75feefd76dc76fb94799ed2756eeda7da14c)